### PR TITLE
Feature/update to polkadot v0 9 22

### PR DESCRIPTION
**Pull request by AndreiNavoichyk**
Updated Substrate to [polkadot-0.9.22](https://github.com/paritytech/substrate/tree/polkadot-v0.9.22)

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -954,6 +959,7 @@ impl pallet_contracts::Config for Runtime {
 	type DeletionWeightLimit = DeletionWeightLimit;
 	type Schedule = Schedule;
 	type AddressGenerator = pallet_contracts::DefaultAddressGenerator;
+	type ContractAccessWeight = pallet_contracts::DefaultContractAccessWeight<RuntimeBlockWeights>;
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:962]: Test SC against this changes.

File: bin/node/cli/tests/running_the_node_and_interrupt.rs [hunk 2]
```diff
@@ -93,7 +93,7 @@ async fn running_two_nodes_with_the_same_ws_port_should_work() {
 	let stderr = first_node.stderr.take().unwrap();
 	let (ws_url, _) = common::find_ws_url_from_output(stderr);
 
-	common::wait_n_finalized_blocks(3, 30, &ws_url).await.unwrap();
+	common::wait_n_finalized_blocks(3, 45, &ws_url).await.unwrap();
```
Comment (MRamanenkau) [inline bin/node/cli/tests/running_the_node_and_interrupt.rs:99]: Why the test is updated?

File: client/finality-grandpa/rpc/src/lib.rs [hunk 3]
```diff
@@ -313,6 +313,8 @@ mod tests {
 	}
 
 	#[tokio::test]
+	// Will be removed in commit https://github.com/paritytech/substrate/commit/abb80e54aa260514a4f12712a9ff8ef944e3b4ec
+	#[ignore]
```
Comment (MRamanenkau) [inline client/finality-grandpa/rpc/src/lib.rs:316]: Could you please add the version where it will be removed?

**Comment by khssnv** [comment]
Is it intentional to have `feature/update-to-polkadot-v0-9-20` as a target instead of the `dev`?

File: bin/node/runtime/Cargo.toml [hunk 4]
```diff
@@ -70,8 +70,6 @@ pallet-indices = { version = "4.0.0-dev", default-features = false, path = "../.
 pallet-identity = { version = "4.0.0-dev", default-features = false, path = "../../../frame/identity" }
 pallet-membership = { version = "4.0.0-dev", default-features = false, path = "../../../frame/membership" }
 pallet-multisig = { version = "4.0.0-dev", default-features = false, path = "../../../frame/multisig" }
-pallet-nomination-pools = { version = "1.0.0", default-features = false, path = "../../../frame/nomination-pools"}
```
Comment (khssnv) [inline bin/node/runtime/Cargo.toml:73]: Is it by intention?

Comment (khssnv) [reply bin/node/runtime/Cargo.toml:73]: There is the dependency deletion, but no pallet configuration deletion in `lib.rs` by the way.

**Review by khssnv — DISMISSED** [review]

File: bin/node/cli/tests/running_the_node_and_interrupt.rs (hunk 2 above)
Comment (AndreiNavoichyk) [reply bin/node/cli/tests/running_the_node_and_interrupt.rs:99]: Added comment.

File: client/finality-grandpa/rpc/src/lib.rs (hunk 3 above)
Comment (AndreiNavoichyk) [reply client/finality-grandpa/rpc/src/lib.rs:316]: Version added here and in 2nd the same test.

**Comment by AndreiNavoichyk** [comment]
> Is it intentional to have `feature/update-to-polkadot-v0-9-20` as a target instead of the `dev`?

Yep as we did not merge v0.9.20 to `master-cere`.

File: bin/node/runtime/Cargo.toml (hunk 4 above)
Comment (AndreiNavoichyk) [reply bin/node/runtime/Cargo.toml:73]: Yes. Pallet configuration deletion has been executed in the previous version.

**Review by MRamanenkau — DISMISSED** [review]

**Review by MRamanenkau — APPROVED** [review]

**Pull request merged**